src/hotspot/cpu/aarch64/aarch64_neon_ad.m4

```diff
@@ -864,119 +864,97 @@ VECTOR_EXTRACT_F(2, D, X, D)
 dnl
 
 // ------------------------------ Vector comparison ---------------------------------
-define(`VECTOR_CMP_EQ_GT_GE', `
-instruct vcm$1$2$3`'(vec$4 dst, vec$4 src1, vec$4 src2, immI cond)
+
+source %{
+
+static Assembler::SIMD_Arrangement elemBytes_to_Arrangement(int esize, bool isX) {
+  switch(esize) {
+    case 1:
+      return isX ? Assembler::T16B : Assembler::T8B;
+    case 2:
+      return isX ? Assembler::T8H : Assembler::T4H;
+    case 4:
+      return isX ? Assembler::T4S : Assembler::T2S;
+    case 8:
+      assert(isX, "unsupported");
+      return Assembler::T2D;
+    default:
+      assert(false, "unsupported");
+      ShouldNotReachHere();
+  }
+  return Assembler::INVALID_ARRANGEMENT;
+}
+
+static void neon_compare(C2_MacroAssembler masm, FloatRegister dst, BasicType bt,
+                         FloatRegister src1, FloatRegister src2, int cond, bool isX) {
```

Comment:
Passing masm by value, rather than reference, is unconventional. There's no reason to do that.

Reply:
Come to think about it, why isn't this an Assembler method?

Reply:
The latest fix moves it into macroAssembler, it's more reasonable.